equinox/nn/_linear.py

```diff
@@ -23,6 +24,7 @@ def __init__(
         in_features: Union[int, Literal["scalar"]],
         out_features: Union[int, Literal["scalar"]],
         use_bias: bool = True,
+        dtype=None,
```

Comment:
The dtype of dtype is missing here. I don't have my laptop for a couple of days but I think the dtype of dtype is jax.numpy.dtype. And it should also be something like "dtype: Optional[jax.numpy.dtype] = None"